### DISCUSSION TITLE
Access package configurations using `::` scoped names

### DISCUSSION THREAD

**Discussion by bdsoha** [laravel/framework#50178] [Ideas]
### Laravel Version

10.4.1

### PHP Version

8.3

### Description

Add the ability to access configurations loaded by a package with *scoped* naming.
For example, assuming the package `mypackage`, I would be able to access configs defined using `mergeConfigFrom(__DIR__ . '/../config.php', 'mypackage')` with `config('mypackage::some-config')`.

This is already available for the following resources:

- **Blade:** `componentNamespace('Components', 'nightshade')` is accessible using `<x-nightshade::calendar />`.
- **Language files:** `loadTranslationsFrom('./lang', 'courier')` is accessible using `trans('courier::messages.welcome')`.
- **Routes:** `Route::name('courier::')->group(...)` is accessible using `routes('courier::welcome')`.
- **Views:** `loadViewsFrom('./resources/views', 'courier')` is accessible using `view('courier::dashboard')`.

### Steps To Reproduce

Described above

**Comment by henzeb** [comment]
The config is merged with key 'mypackage' by the package maintainer in a service provider and will start with that name as the first key. So you can do `config('mypackage.some-config') . why change that behaviour?

**Comment by bdsoha** [comment]
@henzeb I am familiar with the current ability to use a dot *(`.`)* to access the configurations.

Also, I don't want to *change* the existing functionally, instead, I want to have the ability to use `::` to access the configurations.
In the same way I do when accessing *routes*,  *views*, *lang*, etc.

**Reply by henzeb** [reply]
ok, but why do you want that ability? what does it add to the framework? is it just "::" instead of "." or does it do something different?

**Reply by bdsoha** [reply]
It is the *developer experience* of knowing that I am accessing a scoped configuration file, located in an external package or module.
